src/main.rs

```diff
@@ -1,27 +1,63 @@
 #![feature(scoped)]
+#![feature(result_expect)]
 #![allow(deprecated)]
 
+extern crate getopts;
 extern crate librespot;
+extern crate rpassword;
 
 use std::clone::Clone;
 use std::fs::File;
-use std::io::Read;
+use std::io::{stdout, Read, Write};
 use std::path::Path;
 use std::thread;
 use std::path::PathBuf;
 
+use getopts::Options;
+use rpassword::read_password;
+
 use librespot::session::{Config, Session};
 use librespot::util::version::version_string;
 use librespot::player::Player;
 use librespot::spirc::SpircManager;
 
+fn usage(program: &str, opts: &Options) -> String {
+    let brief = format!("Usage: {} [options]", program);
+    format!("{}", opts.usage(&brief))
+}
+
 fn main() {
-    let mut args = std::env::args().skip(1);
-    let mut appkey_file = File::open(Path::new(&args.next().unwrap())).unwrap();
-    let username = args.next().unwrap();
-    let password = args.next().unwrap();
-    let cache_location = args.next().unwrap();
-    let name = args.next().unwrap();
+    let args: Vec<String> = std::env::args().collect();
+    let program = args[0].clone();
+
+    let mut opts = Options::new();
+    opts.reqopt("a", "appkey", "Path to a spotify appkey", "APPKEY");
+    opts.reqopt("u", "username", "Username to sign in with", "USERNAME");
+    opts.optopt("p", "password", "Password (optional)", "PASSWORD");
+    opts.reqopt("c", "cache", "Path to a directory where files will be cached.", "CACHE");
+    opts.reqopt("n", "name", "Device name", "DEVICE");
+    let matches = match opts.parse(&args[1..]) {
+        Ok(m) => { m },
+        Err(f) => { 
+                print!("Error: {}\n{}", f.to_string(), usage(&*program, &opts));
+                return;
+        }
+    };
+
+    let mut appkey_file = File::open(
+                                    Path::new(&*matches.opt_str("a").unwrap())
+                                ).expect("Could not open app key.");
+
+    let username = matches.opt_str("u").unwrap();
+    let cache_location = matches.opt_str("c").unwrap();
+    let name = matches.opt_str("n").unwrap();
+
+    let password = matches.opt_str("p").unwrap_or_else(||
+        {
```

Comment:
Small nit, but could you put the brace on the first line, and reduce the indentation by a tab